d2ir/compile.go

```diff
@@ -459,6 +461,12 @@ func (c *compiler) compileMap(dst *Map, ast, scopeAST *d2ast.Map) {
 						Value: []d2ast.InterpolationBox{{Substitution: n.Substitution}},
 					},
 				},
+				References: []*FieldReference{{
+					Context_: &RefContext{
+						Scope:    ast,
+						ScopeMap: dst,
+					},
```

Comment:
does this need ScopeAST similar to compileKey above?

Reply:
Hm it doesn't right now but better to put it for consistency.